packages/images-plugin/lib/finder.js

```diff
@@ -72,16 +73,19 @@ const DEFAULT_SETTINGS = {
   getMatchedImageResult: false,
 };
 
-export default class ImageElementFinder {
-  /** @type {ExternalDriver} */
+export class ImageElementFinder {
+  /**
+   * Can be set post-instantiation
+   * @type {ExternalDriver|undefined}
+   */
   driver;
 
   /** @type {LRU<string,ImageElement>} */
   imgElCache;
 
   /**
    *
-   * @param {ExternalDriver} driver
+   * @param {ExternalDriver} [driver]
```

Comment:
why brackets are needed around param names?

Reply:
this means that the params are optional. they can be `undefined`. they cannot be `null`, however; that'd look like `@param {ExternalDriver?} [driver]` which allows both `null` and `undefined`, or `@param {ExternalDriver?} driver` which wants an `ExternalDriver` or `null` but not `undefined`.

Reply:
oh, that's getting too complicated IMHO. Should  we really care about differentiating undefined and null except of some (rare) special cases?

Reply:
well, we kind of have to, because they are different types.  I'm not sure there's any way for TS to treat them as the same via config.  

rant time!

There's no consensus around use of `undefined` vs `null` in JS. _Mostly_, I've seen `null` avoided entirely (I generally subscribe to this philosophy), since (again, _mostly_) using one or the other is simply a matter of convention (aside: the only place where `null` _must_ be used is `Object.create(null)`).   In other words, there's no inherent meaning in the use of `null` other than what the developer ascribes to it.

For example, if the type of `sessionId` in `BaseDriver` was `string|undefined` instead of `string|null`, it'd be functionality equivalent (though `sessionId === null` comparisons will need to be replaced with `sessionId === undefined`).  Anything (within reason) you can do with a `sessionId` of `null` you can do with a `sessionId` of `undefined`.

What's problematic: when there really _isn't_ a convention, and sometimes things are `null` and it's unclear why they aren't just `undefined`, or vice-versa.  😜